src/Diagnostics.RuntimeHost/Controllers/InternalAPIController.cs

```diff
@@ -0,0 +1,163 @@
+                    break;
+                case "TrainingException":
+                    var trainingException = JsonConvert.DeserializeObject<InternalAPITrainingException>(eventContent);
+                    DiagnosticsETWProvider.Instance.LogInternalAPITrainingException(trainingException.RequestId, trainingException.TrainingId, trainingException.ProductId, trainingException.ExceptionType, trainingException.ExceptionDetails);
+                    break;
+                case "Insights":
+                    var insights = JsonConvert.DeserializeObject<InternalAPIInsights>(eventContent);
+                    DiagnosticsETWProvider.Instance.LogInternalAPIInsights(insights.RequestId, insights.Message);
+                    break;
+                case "APISummary":
+                    var apiSummary = JsonConvert.DeserializeObject<InternalAPISummary>(eventContent);
+                    DiagnosticsETWProvider.Instance.LogInternalAPISummary(apiSummary.RequestId, apiSummary.OperationName, apiSummary.StatusCode, apiSummary.LatencyInMilliseconds, apiSummary.StartTime, apiSummary.EndTime, apiSummary.Content);
+                    break;
+                case "TrainingSummary":
+                    var trainingSummary = JsonConvert.DeserializeObject<InternalAPITrainingSummary>(eventContent);
+                    DiagnosticsETWProvider.Instance.LogInternalAPITrainingSummary(trainingSummary.RequestId, trainingSummary.TrainingId, trainingSummary.ProductId, trainingSummary.LatencyInMilliseconds, trainingSummary.StartTime, trainingSummary.EndTime, trainingSummary.Content);
+                    break;
+                default:
+                    DiagnosticsETWProvider.Instance.LogInternalAPIMessage(eventContent);
+                    break;
+            }
+
+            return Ok();
+        }
+
+        /// <summary>
+        /// Uploads model files to github
+        /// </summary>
+        /// <param name="modelpath">Model path</param>
+        [HttpPost(UriElements.PublishModel)]
+        public async Task<IActionResult> PublishModel(string trainingId, [FromBody]string modelpath)
+        {
+            var commits = _internalApiHelper.GetAllFilesInFolder(modelpath);
+            var watcher = _sourceWatcherService.Watcher as GitHubWatcher;
+            await watcher._githubClient.CreateOrUpdateFiles(commits, $"Model update trainingId {trainingId}");
+            return Ok("Model published Successfully");
+        }
+
+        [HttpPost(UriElements.RefreshModel)]
+        public async Task<IActionResult> RefreshModel(string productId)
+        {
+            Request.Headers.TryGetValue(HeaderConstants.RequestIdHeaderName, out StringValues requestIds);
+            var requestId = requestIds.FirstOrDefault();
+            var parameters = new Dictionary<string, string>();
+            parameters.Add("productId", productId);
```

Comment:
What does `productId` represent?

Reply:
its the PesIds like windows is 14748 etc..